### PR TITLE
Disk/Partition: add support for /dev/mapper devices and partitions

### PR DESCRIPTION
**Pull request by stdweird**

### CODE REVIEW

File: src/main/perl/BlockdevFactory.pm [hunk 1]
```diff
@@ -7,27 +7,60 @@ NCM::BlockdevFactory
...
+use Readonly;
+
+Readonly my $BASEPATH => "/system/blockdevices";
+Readonly my @PARTED	=> qw (/sbin/parted -s --);
+Readonly my @PARTEDEXTRA => qw (u MiB);
+Readonly my $PARTEDPRINT => 'print';
+# if all is put in qw(), perl wans with 'Possible attempt to separate words with commas'
+Readonly my $DMINFO => [qw(dmsetup info -C --noheadings --separator : -o), 'name,subsystem'];
+
+Readonly my %BUILD_MAP => (
```
Comment (ned21) [inline src/main/perl/BlockdevFactory.pm]: This seems to control which modules are loaded for a given PAN path. Can this be documented explicitly to make it easy to see how to add new types?

**Comment by stdweird** [comment]
@ned21 i don't get the `LVM` -> `VG` remark. the old `LVM` module handled `LVM` volume groups; and there's also a `LV` module to handle `LVM` logical volumes. so i renamed `LVM` in `VG`.
i can move them in a namespace `LVM::VG` and `LVM::LV` if needed.

**Comment by stdweird** [comment]
@ned21 i added a comment to clarify the BUILD_MAP

**Comment by ned21** [comment]
@stdweird Got it, thanks. So e99439b should say:

```
Rename NCM::LVM to NCM::VG

This makes it consistent with name of the NCM::LV module used
to configure LVM Logical Volumes.
```

File: src/main/perl/BlockdevFactory.pm [hunk 2]
```diff
@@ -7,27 +7,66 @@ NCM::BlockdevFactory
...
+Readonly my @PARTEDEXTRA => qw (u MiB);
+Readonly my $PARTEDPRINT => 'print';
+# if all is put in qw(), perl warns with 'Possible attempt to separate words with commas'
+Readonly my $DMINFO => [qw(dmsetup info -C --noheadings --separator : -o), 'name,subsystem'];
+
+# Pattern or compiled pattern to map the device taken fom the
+# pan path /system/blockdevices/<device> to the NCM:: class
+# If the pattern is a string, it is a partial pattern
+# (actual used one is '^<string>/')
+# Patterns are tried in alphabetical order,
```
Comment (ned21) [inline src/main/perl/BlockdevFactory.pm]: ```alphabetical order of the key name,``` (if I have read the later code correctly)

**Comment by stdweird** [comment]
@ned21 remarks addressed

**Comment by jrha** [comment]
@stdweird I still can't see jenkins0 so have no idea why the tests are failing — could you take a look?

**Comment by stdweird** [comment]
retest this please
@jrha failed coverage test due to jenkins misconfig

**Pull request merged**